lib/unordered-equal.js

```diff
@@ -0,0 +1,22 @@
+export const checkValueForUnorderedEqual = value => {
+	/* eslint-disable indent, operator-linebreak, unicorn/no-nested-ternary */
+	const type = (
+		value instanceof Map ? 'map' :
+		value instanceof Set ? 'set' :
+		Array.isArray(value) ? 'array' :
+		'invalid'
+	);
+	/* eslint-enable indent, operator-linebreak, unicorn/no-nested-ternary */
```

Comment:
This probably does read better as a series of `if` statements that are assigning to a let-bound `type` variable.